pkg/ccl/backupccl/file_sst_sink_test.go

```diff
@@ -135,3 +141,819 @@ func TestFileSSTSinkExtendOneFile(t *testing.T) {
+				{
+					input: []kvAndTS{
+						{key: "a", value: []byte("1"), timestamp: 10},
+						{key: "c", value: []byte("2"), timestamp: 10},
+					},
+				},
+				{
+					input: []kvAndTS{
+						{key: "b", value: []byte("3"), timestamp: 10},
+						{key: "d", value: []byte("4"), timestamp: 10},
+					},
+					expectErr: "keys must be added in strictly increasing order",
+				},
+			},
+		},
+		{
+			name: "out-of-order-timestamp",
+			inputs: []testInput{
+				{
+					input: []kvAndTS{
+						{key: "a", value: []byte("1"), timestamp: 10},
+						{key: "b", value: []byte("2"), timestamp: 10},
+					},
+				},
+				{
+					input: []kvAndTS{
+						{key: "b", value: []byte("3"), timestamp: 10},
+						{key: "c", value: []byte("4"), timestamp: 10},
+					},
+					expectErr: "keys must be added in strictly increasing order",
+				},
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			buf := &bytes.Buffer{}
+			sst := storage.MakeBackupSSTWriter(ctx, settings, buf)
+			sink := fileSSTSink{sst: sst}
+			compareSST := true
+
+			for _, input := range tt.inputs {
+				kvs := input.input
+				// Add a range key in the input as well.s
+				es := newExportedSpanBuilder(kvs[0].key, kvs[len(kvs)-1].key, false).
+					withKVs(kvs).
+					withRangeKeys([]rangeKeyAndTS{{"a", "z", 10}}).
+					build()
+				err := sink.copyPointKeys(es.dataSST)
+				if input.expectErr != "" {
+					// Do not compare resulting SSTs if we expect errors.
+					require.ErrorContains(t, err, input.expectErr)
+					compareSST = false
+				} else {
+					require.NoError(t, err)
+				}
+			}
+
+			sst.Close()
+
+			if !compareSST {
+				return
+			}
+
+			var expected []kvAndTS
+			for _, input := range tt.inputs {
+				expected = append(expected, input.input...)
+			}
+
+			iterOpts := storage.IterOptions{
+				KeyTypes:   storage.IterKeyTypePointsAndRanges,
+				LowerBound: keys.LocalMax,
+				UpperBound: keys.MaxKey,
+			}
+
+			iter, err := storage.NewMemSSTIterator(buf.Bytes(), false, iterOpts)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			var actual []kvAndTS
+			for iter.SeekGE(storage.MVCCKey{}); ; iter.Next() {
+				if ok, err := iter.Valid(); err != nil {
+					t.Fatal(err)
+				} else if !ok {
+					break
+				}
+
+				key := iter.UnsafeKey()
+				value, err := iter.UnsafeValue()
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				kv := kvAndTS{key: string(key.Key), timestamp: key.Timestamp.WallTime}
+				kv.value = append(kv.value, value...)
+
+				actual = append(actual, kv)
+			}
+
+			require.Equal(t, expected, actual)
+		})
+	}
+}
+
+func TestFileSSTSinkCopyRangeKeys(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	settings := cluster.MakeTestingClusterSettings()
+
+	type testInput struct {
+		input     []rangeKeyAndTS
+		expectErr string
+	}
+
+	type testCase struct {
+		name   string
+		inputs []testInput
+	}
+
+	for _, tt := range []testCase{
+		{
+			name: "single-sst",
```

Comment:
nit: i wonder if it would be more clear to name these tests "single-exported-span" and "multiple-exported-span" since the file sink writes multiple of these exportedSpans (i.e. backup  manifest file entries) to one physical sst.

Reply:
sam comment applies to the point key test